tests/monotouch-test/Compression/ThoroughCompressionStreamTest.cs

```diff
@@ -0,0 +1,121 @@
+ 		string lz4CompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "compressed_lz4");
+ 		string lzeCompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "compressed_lze");
+ 		string lzeUncompressedFilePath = Path.Combine (NSBundle.MainBundle.BundlePath, "uncompressed.txt");
+#endif
+
+		string firstTempPath;
+		string secondTempPath;
+
+		private static void CopyStream (Stream src, Stream dest)
+		{
+			byte[] array = new byte[1024];
+			int bytes_read;
+			bytes_read = src.Read (array, 0, 1024);
+			while (bytes_read != 0) {
+				dest.Write (array, 0, bytes_read);
+				bytes_read = src.Read (array, 0, 1024);
+			}
+		}
+
+		private static bool compare_buffers (byte[] first, byte[] second, int length)
+		{
+			if (first.Length < length || second.Length < length) {
+				return false;
+			}
+			for (int i = 0; i < length; i++) {
+				Console.WriteLine (first[i]);
+				Console.WriteLine (second[i]);
+				if (first[i] != second[i]) {
+					return false;
+				}
+			}
+			return true;
+		}
+
+		[SetUp]
+		public void SetUp ()
+		{
+			firstTempPath = Path.GetTempFileName ();
+			secondTempPath = Path.GetTempFileName ();
+		}
+
+		void DecodeRealFile (CompressionAlgorithm algorithm, string compressedFile, string uncompressedFile)
+		{
+			Console.WriteLine ("Decoding");
+			var dataStream = File.OpenRead (compressedFile);
+			var backing = File.Create (firstTempPath);
+			backing.Seek (0, SeekOrigin.Begin);
+			CompressionStream decompressing = new CompressionStream (dataStream, CompressionMode.Decompress, algorithm);
+			MemoryStream output = new MemoryStream ();
+			CopyStream (decompressing, output);
+			Console.WriteLine ($"Lenght is {output.Length}");
+			Assert.IsTrue (compare_buffers (File.ReadAllBytes (uncompressedFile), output.GetBuffer(), (int) output.Length));
+			Console.WriteLine ("Assertion positive.");
+			output.Seek (0, SeekOrigin.Begin);
+			using (StreamReader reader = new StreamReader(output))
+			{
+				Console.WriteLine("Reading all");
+				Console.WriteLine (reader.ReadToEnd());
+			}
+			decompressing.Close();
+			output.Close();
+		}
+
+		[TestCase (CompressionAlgorithm.LZ4)]
+		[TestCase (CompressionAlgorithm.Lzfse)]
+		[TestCase (CompressionAlgorithm.Lzma)]
+		[TestCase (CompressionAlgorithm.Zlib)]
+		public void TestDecodeRealFile (CompressionAlgorithm algorithm)
+		{
+			switch (algorithm)
+			{
+			case CompressionAlgorithm.LZ4:
+				DecodeRealFile (algorithm, lz4CompressedFilePath, lzeUncompressedFilePath);
+			break;
+			case CompressionAlgorithm.Lzfse:
+				DecodeRealFile (algorithm, lzeCompressedFilePath, lzeUncompressedFilePath);
+			break;
+			case CompressionAlgorithm.Lzma:
+				DecodeRealFile (algorithm, lzmaCompressedFilePath, lzeUncompressedFilePath);
+			break;
+			case CompressionAlgorithm.Zlib:
+				DecodeRealFile (algorithm, zipCompressedFilePath, lzeUncompressedFilePath);
+			break;
```

Comment:
your code styles is off, double check your settings

Also it looks simpler to have a method per algorithm since the code is different in each case.

Or provide the file name (without the path) in the `[TestCase]` arguments and share the code (+ include the path calculation).

Reply:
I was considering the second one, pass the path name, and deal with it accordingly.